src/odin/directionsbuilder.cc

```diff
@@ -62,43 +62,49 @@ const std::unordered_map<int, DirectionsLeg_TravelMode> translate_travel_mode{
     {static_cast<int>(TripLeg_TravelMode_kTransit), DirectionsLeg_TravelMode_kTransit},
 };
 
-DirectionsBuilder::DirectionsBuilder() {
-}
-
 // Returns the trip directions based on the specified directions options
 // and trip path. This method calls ManeuversBuilder::Build and
 // NarrativeBuilder::Build to form the maneuver list. This method
 // calls PopulateDirectionsLeg to transform the maneuver list into the
 // trip directions.
-DirectionsLeg DirectionsBuilder::Build(const DirectionsOptions& directions_options,
-                                       TripLeg& trip_path) {
-  // Validate trip path node list
-  if (trip_path.node_size() < 1) {
-    throw valhalla_exception_t{210};
-  }
+void DirectionsBuilder::Build(Api& api) {
```

Comment:
this looks like a lot of change but its all mechanical. we pass odin the full pbf and it fills in all the legs by looping over them.